pyscf/scf/dispersion.py

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env python
+# Copyright 2014-2023 The PySCF Developers. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Author: Xiaojie Wu <wxj6000@gmail.com>
+#
+
+'''
+dispersion correction for HF and DFT
+'''
+
+
+import numpy
+from pyscf.scf.hf import KohnShamDFT
+
+def get_dispersion(mf, disp_version=None):
+    if disp_version is None:
+        disp_version = mf.disp
+    mol = mf.mol
+    if disp_version is None:
+        return 0.0
+    if isinstance(mf, KohnShamDFT):
+        method = mf.xc
+    else:
+        method = 'hf'
+
+    # for dftd3
+    if disp_version[:2].upper() == 'D3':
```

Comment:
`disp_version` may be `None`, as the default value in `__init__` of SCF object. Similar problems are in the grad, hessian and other relevant modules.

Reply:
Right. `disp_version` is `None` by default. That is, there is no dispersion correction by default. The dispersion correction is added when `disp` is assigned to `mf` explicitly, such as `mf.disp = "d3bj"`.